src/components/_internal/BaseButton/BaseButton.tsx

```diff
@@ -108,6 +108,7 @@ const BaseButton: React.FC<
       $isLoading={isLoading}
       $margin={margin}
       $variant={variant}
+      aria-label={props['aria-label']}
```

Comment:
So this actually doesn't solve the issue described in UXD-1014. The thing is that we need to provide our own `aria-label` for cases when the button has `isLoading === true && hasOnlyIcon === true`. In other cases, it is up to the consumer to provide `aria-label`. We should also move the line under the `{...props}` so this won't get overridden by the user-passed `aria-label` property.

```suggestion
      aria-label={isLoading && hasOnlyIcon ? 'Loading' : props?.['aria-label']}
```